uaclient/config.py

```diff
@@ -598,40 +611,26 @@ def process_config(self):
                 )
             )
 
-    def write_cfg(self, config_path=None):
-        """Write config values back to config_path or DEFAULT_CONFIG_FILE."""
-        if not config_path:
-            config_path = DEFAULT_CONFIG_FILE
-        content = messages.UACLIENT_CONF_HEADER
-        cfg_dict = copy.deepcopy(self.cfg)
-        if "log_level" not in cfg_dict:
-            cfg_dict["log_level"] = CONFIG_DEFAULTS["log_level"]
-        # Ensure defaults are present in uaclient.conf if absent
-        for attr in (
-            "contract_url",
-            "security_url",
-            "data_dir",
-            "log_file",
-            "timer_log_file",
-            "daemon_log_file",
-        ):
-            cfg_dict[attr] = getattr(self, attr)
-
-        # Each UA_CONFIGURABLE_KEY needs to have a property on UAConfig
-        # which reads the proper key value or returns a default
-        cfg_dict["ua_config"] = {
-            key: getattr(self, key, None) for key in UA_CONFIGURABLE_KEYS
-        }
-
-        content += yaml.dump(cfg_dict, default_flow_style=False)
-        system.write_file(config_path, content)
-
     def warn_about_invalid_keys(self):
         if self.invalid_keys is not None:
             for invalid_key in sorted(self.invalid_keys):
                 logging.warning(
                     "Ignoring invalid uaclient.conf key: %s", invalid_key
                 )
+        if "ua_config" in self.cfg:
```

Comment:
Should we have an integration test for this ? I think we will show this to the user when running a CLI command because we have change the log level to debug, which should overwrite `disable_log_to_console`. But it is worth having a test confirming that

Reply:
The console log level is hard-coded to INFO unless you call the cli with --debug, so I think this should behave the same as the existing warning.

It's still probably a good idea to have an integration test specifically for supporting the old `ua_config` and asserting what the output will be. I'll add it

Reply:
Added!